master/buildbot/util/service.py

```diff
@@ -186,6 +186,8 @@ class AsyncService(service.Service):
     def setServiceParent(self, parent):
         if self.parent is not None:
             yield self.disownServiceParent()
+        # XXX(sa2ajj): why this magic?  The twisted version of setServiceParent
+        # does not lookg this complicated (the line below)
         parent = service.IServiceCollection(parent, parent)
```

Comment:
This is the magic of zope.interface and the adapters.
It will verify that parent is a IServiceCollection or that it can be adapted to an IServiceCollection